### PR TITLE
timePicker

### PR DESCRIPTION
**Pull request by eliasreis54**

### CODE REVIEW

**Review by thales-goncalves — COMMENTED** [review]
Commit ficou com o print('eliias'); no _selectTime

File: lib/presentation/pages/create_task_page.dart [hunk 1]
```diff
@@ -25,17 +25,27 @@ class CreateTaskPage extends StatelessWidget {
     );
   }
 
+  Future _selectTime(BuildContext context) async {
+    TimeOfDay _currentTime = new TimeOfDay.now();
+    print('eliias');
```
Comment (thales-goncalves) [inline lib/presentation/pages/create_task_page.dart]: Remover print, por favor

Comment (eliasreis54) [reply lib/presentation/pages/create_task_page.dart]: Opa! Acabou passando. vou ajustar!

File: lib/presentation/pages/create_task_page.dart [hunk 2]
```diff
@@ -25,17 +25,26 @@ class CreateTaskPage extends StatelessWidget {
...
   }
 
+  Future _selectTime(BuildContext context) async {
+    TimeOfDay _currentTime = new TimeOfDay.now();
+    TimeOfDay selectedTime = await showTimePicker(
+        context: context,
+        initialTime: _currentTime,
+    );
+    if(selectedTime != null) print(selectedTime.toString());
+  }
```
Comment (mugbug) [inline lib/presentation/pages/create_task_page.dart]: What do you think of moving the responsibility of presenting the picker to the component? I think it makes more sense

Comment (eliasreis54) [reply lib/presentation/pages/create_task_page.dart]: I agree with you. 
I did it, and in this case, the same component can render different types, i made a enum for this.

**Review by thales-goncalves — APPROVED** [review]

File: lib/presentation/components/field_picker.dart [hunk 3]
```diff
@@ -1,38 +1,73 @@
...
+      case FieldPickerTypes.remember: {
+        print('remaider');
+      }
+      break;
+    }
   }
+
+    @override
+    Widget build(BuildContext context) {
+      return GestureDetector(
```
Comment (mugbug) [inline lib/presentation/components/field_picker.dart]: Acho que se vc usar InkWell, o card dá um feedback de tap, estilo material. Acho que fica legal.

File: lib/presentation/components/field_picker.dart [hunk 4]
```diff
@@ -1,38 +1,73 @@
...
 import 'package:healthy_routine_mobile/resources/styles/colors.dart';
+import 'package:healthy_routine_mobile/healthy_routine.dart';
+
 
 class FieldPicker extends StatelessWidget {
   final IconData icon;
   final String label;
   final String value;
+  final FieldPickerTypes type;
+  final Function(TimeOfDay) onTap;
```
Comment (mugbug) [inline lib/presentation/components/field_picker.dart]: This block won't be called on tap. I'd change it to onChange or something, what do you think?

File: lib/presentation/components/field_picker_types.dart [hunk 5]
```diff
@@ -0,0 +1,5 @@
+enum FieldPickerTypes {
```
Comment (mugbug) [inline lib/presentation/components/field_picker_types.dart]: I prefer FieldPickerType, as singular once it's a type.

File: lib/presentation/components/field_picker_types.dart [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+  time,
+  repeat,
+  remember,
```
Comment (mugbug) [inline lib/presentation/components/field_picker_types.dart]: Suggestion: reminder

File: lib/presentation/components/field_picker.dart (hunk 4 above)
Comment (eliasreis54) [reply lib/presentation/components/field_picker.dart]: Yep
Done

File: lib/presentation/components/field_picker_types.dart (hunk 6 above)
Comment (eliasreis54) [reply lib/presentation/components/field_picker_types.dart]: Done

File: lib/presentation/components/field_picker_types.dart (hunk 5 above)
Comment (eliasreis54) [reply lib/presentation/components/field_picker_types.dart]: Make sense.

File: lib/presentation/components/field_picker.dart (hunk 3 above)
Comment (eliasreis54) [reply lib/presentation/components/field_picker.dart]: Alterei, precisa dar um tapa na animação depois, mas ta funcionando.

**Review by thales-goncalves — APPROVED** [review]

**Pull request merged**